packages/gatsby-theme-aio/algolia/create-algolia-record.js

```diff
@@ -10,135 +10,37 @@
-    return description;
-  }
-
-  // TODO: Replace with paragraph content from record.
-  function getContent(rawRecord, file) {
-    const content = rawRecord.content || file.description || file.excerpt || '';
-    return content;
-  }
-
-  // TODO: Add reporter for keywords to create frontmatter report for Adobe teams.
-  function getFileKeywords(file) {
-    return file.keywords;
-  }
-
-  // TODO: Reduce cardinality for use in search customRanking setting
-  function getFileWordCount(file) {
-    const words = file.words || 0;
-    return words;
-  }
-
-  // TODO: Reduce cardinality for use in search customRanking setting
-  function getFileSize(file) {
-    const size = file.size;
-    return size;
-  }
-
-  function getFileModifiedDate(file) {
-    const lastUpdated = file.lastUpdated;
-    return lastUpdated;
-  }
-
-  // TODO: Either expand this or remove and just use file.slug
-  function getFileSlug(file) {
-    const slug = file.slug || '';
-    return slug;
-  }
-
-  // Gets all headings above the rawRecord's content
-  function getHeadings(rawRecord, file) {
-    let headingsAboveRawRecord = selectAll('heading', file.mdxAST);
-
-    const goodHeadings = headingsAboveRawRecord.filter(
-      ({ position }) => position.start.line < rawRecord.positionEndLine
-    );
-
-    // Removes jsdoc code tabs
-    const betterHeadings = goodHeadings.filter(
-      goodHeading =>
-        goodHeading.children[0].value !== 'Request' && goodHeading.children[0].value !== 'Response'
-    );
-
-    const headings = betterHeadings.map(heading => heading.children[0].value);
-    return headings;
-  }
-
-  // Gets the closest heading above the rawRecord's content
-  function getContentHeading(rawRecord, file) {
-    const contentHeading = getHeadings(rawRecord, file).slice(-1)[0];
-    return contentHeading;
-  }
-
-  // The anchor is just the transformed contentHeading
-  function getAnchor(rawRecord, file) {
-    const contentHeading = getContentHeading(rawRecord, file);
-    if (contentHeading == null) return '';
-
-    // TODO: Handle headings with custom anchors from
-    //  the gatsby-remark-autolink-headers plugin
-    return `#${contentHeading
-      .match(/[a-zA-Z0-9]\w+/g)
-      ?.map(s => s.toLowerCase())
-      .join('-')}`;
-  }
+  return record;
+}
 
-  function getUrl(rawRecord, file) {
-    const url = `${process.env.GATSBY_SITE_DOMAIN_URL}${process.env.PATH_PREFIX}${
-      file.slug == null ? '' : file.slug
-    }${getAnchor(rawRecord, file)}`;
-    return url;
-  }
+function getUrl(rawRecord, file) {
+  const url = `${process.env.GATSBY_SITE_DOMAIN_URL}${process.env.PATH_PREFIX}${
+    file.slug == null ? '' : file.slug
+  }${rawRecord.anchor}`;
+  return url;
+}
 
-  function getSpotlight(rawRecord, file) {
-    const spotlight = file.spotlight;
-    return spotlight;
-  }
+function getDateLastUpdated(file) {
+  const lastUpdated = file.lastUpdated;
```

Comment:
May need to do:
file?.lastUpdated

For whatever reason, I noticed the last updated to be pretty janky on docs sometimes and it returns undefined.

Reply:
Yeah, this is a problem with the html source files coming from the html sources (iframes, imports, and openapis). right now, i'm just using the host/parent's lastUpdated date. It's the best we can do at the moment. 

But with the rewrite, the file itself will never be undefined or null because the caller is in a loop controlled by the markdownFiles:

```js
        for (const markdownFile of markdownFiles) {
        ...
          for (const rawRecord of rawRecords) {
            const record = await createAlgoliaRecord(rawRecord, markdownFile);
```

Reply:
Ah ok.